core/server/api_container/server/startosis_engine/kurtosis_instruction/kurtosis_instruction.go

```diff
@@ -34,5 +35,11 @@ type KurtosisInstruction interface {
 	ValidateAndUpdateEnvironment(environment *startosis_validator.ValidatorEnvironment) error
 
 	// TryResolveWith assesses whether the instruction can be resolved with the one passed as an argument.
-	TryResolveWith(other KurtosisInstruction, enclaveComponents *enclave_structure.EnclaveComponents) enclave_structure.InstructionResolutionStatus
+	TryResolveWith(other EnclavePlanInstruction, enclaveComponents *enclave_structure.EnclaveComponents) enclave_structure.InstructionResolutionStatus
+
+	GetCapabilites() EnclavePlanCapabilitiesProvider
+}
+
+type EnclavePlanCapabilitiesProvider interface {
```

Comment:
nit: one interface per file maybe?

Reply:
What's this purpose of this provider? Can't the above `GetCapabilities` return the capability straight, whithout this provider wrapper?